### PR TITLE
add notifications of switch status

### PR DESCRIPTION
**Pull request by akash1810**
Notify via [Anghammarad](https://github.com/guardian/anghammarad) when:
- switch has been changed (via email and chat)
- switch is still active (via chat)

### CODE REVIEW

**Comment by akash1810** [comment]
This isn't working in CODE for an unknown reason. Parking for now.

File: app/config/Switches.scala [hunk 1]
```diff
@@ -53,7 +56,7 @@ class Switches(config: LoginConfig) {
   def start() {
     Logger.info("Starting switches scheduled task")
 
-    scheduler.scheduleAtFixedRate(() => refresh(), 60, 60, TimeUnit.SECONDS)
+    scheduler.scheduleAtFixedRate(() => refresh(), 0, 60, TimeUnit.SECONDS)
```
Comment (sihil) [inline app/config/Switches.scala]: This is much better.

File: app/config/Switches.scala [hunk 2]
```diff
@@ -70,6 +73,8 @@ class Switches(config: LoginConfig) {
       val source = Source.fromInputStream(result.getObjectContent).mkString
       val statesInS3 = Json.parse(source).as[Map[String, SwitchState]]
 
+      statesInS3.filter(_._2 == On).keys.foreach(notifier.sendStillActiveNotification)
+
```
Comment (sihil) [inline app/config/Switches.scala]: This happens every minute right? That seems obnoxious. Every hour might be OK.

Comment (sihil) [reply app/config/Switches.scala]: Perhaps by something as simple as `DateTime.now().minute() % 0 == 0`?

File: app/config/LoginConfig.scala [hunk 3]
```diff
@@ -31,7 +41,9 @@ object LoginConfig {
 
    val switchBucket = "login-gutools-config"
 
-    LoginConfig(stage, domain, host, appName, emergencyAccessTableName, tokensTableName, tokenReissueUri, emailSettings, switchBucket)
+   val anghammaradSnsArn = "arn:aws:sns:eu-west-1:095768028460:anghammarad-PROD-NotificationTopic-HDJHBGZT0FFD"
```
Comment (sihil) [inline app/config/LoginConfig.scala]: Indent? Lucky it's not 🐍

File: app/config/Switches.scala (hunk 2 above)
Comment (akash1810) [reply app/config/Switches.scala]: Good idea. Done.

File: app/config/Switches.scala [hunk 4]
```diff
@@ -70,6 +76,12 @@ class Switches(config: LoginConfig, s3Client: AmazonS3) extends Loggable {
       val source = Source.fromInputStream(result.getObjectContent).mkString
       val statesInS3 = Json.parse(source).as[Map[String, SwitchState]]
 
+      // we're refreshing every minute, only sendStillActiveNotification once an hour
+      val isOnTheHour = Try(DateTime.now().getMinuteOfHour % 0 == 0).getOrElse(false)
```
Comment (sihil) [inline app/config/Switches.scala]: This might bury a problem. What exception are you guarding against here?

File: app/utils/Notifier.scala [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+  private def sendNotification(message: String, channel: RequestedChannel): String = {
+    log.info("talking to Anghammarad")
+
+    Await.result(Anghammarad.notify(
+      subject = "login.gutools switches monitor",
+      message = message,
+      actions = List(
+        // yeah, this is gross, we need better runbooks!
+        Action("Open runbook", "https://github.com/guardian/login.gutools#emergency-access-when-google-auth-is-down")
```
Comment (sihil) [inline app/utils/Notifier.scala]: Another action could take you to the login tool's switch page once #40 is merged

File: app/config/Switches.scala (hunk 4 above)
Comment (akash1810) [reply app/config/Switches.scala]: a `java.lang.ArithmeticException` can be thrown with `DateTime.now().getMinuteOfHour % 0`

Comment (sihil) [reply app/config/Switches.scala]: That's because I'm a muppet. You can't `% 0`. This can just be `getMinuteOfHour == 0`. 🤦‍♂

Comment (sihil) [reply app/config/Switches.scala]: Or perhaps it is worth making a one hour scheduler that is less of a hack....

**Comment by sihil** [comment]
Super elegant. 💯

**Pull request merged**